js/gacha.js

```diff
@@ -23,8 +112,34 @@ function pullCards(cards, n) {
   return samples.slice(0, n);
 }
 
-//Pulls "n" number of cards from the cards_data array and renders them in CARD_LIST.
-export function pullAndRenderCards(cards_data, n, render_location) {
-  let pulled = pullCards(cards_data, n);
+//Pulls cards from the cards_data array and renders them in render_location.
+export function pullAndRenderCards(cards_data, render_location) {
+  //let pulled = getRandomCards(cards_data, n);
+  let pulled = pullCards(slots);
   renderCards(pulled, render_location, true);
 }
+
+//Debug function to get the average rates for each rarity over a list of pulls.
+function calculateRates(pulls) {
+  let rates = {};
+  let total = 0;
+  for (let rarity in cards_by_rarity) {
+    rates[rarity] = 0;
+  }
+  for (let pull of pulls) {
+    for (let card of pull) {
+      rates[card["Rarity Folder"]]++;
+      total++;
+    }
+  }
+  for (let rarity in cards_by_rarity) {
+    rates[rarity] /= total;
+    rates[rarity] *= 100;
+  }
+  return rates;
+}
+
+function testRates() {
```

Comment:
I didn't run this myself, but trust that you have o7

Reply:
Ran it again and left the results in a comment below o7